api-reference/10 UI Widgets/dxGantt/1 Configuration/tasks/endExpr.md

```diff
@@ -8,3 +8,141 @@ default: 'end'
 Specifies the data field that provides tasks' end dates.
 
 ---
+---
+##### jQuery
+
+    <!-- tab: index.js -->
+    $(function() {
+        $("#gantt").dxGantt({
+            tasks: {
+                dataSource: tasks,
+                startExpr: "startDate",
+                endExpr: "endDate"
+            },
+            //...
+        });
+    });
+    <!-- tab: data.js -->
+    var tasks = [{
+        'id': 1,
+        'parentId': 0,
+        'title': 'Software Development',
+        'startDate': new Date('2019-02-21T05:00:00.000Z'),
+        'endDate': new Date('2019-07-04T12:00:00.000Z'),
+        'progress': 31
+    },
+    // ...
+    ];
+
+##### Angular
+
+    <!-- tab: app.component.ts -->
+    import { DxGanttModule } from 'devextreme-angular';
+    import { Service, Task, Dependency, Resource, ResourceAssignment } from './app.service';
+    // ...
+    export class AppComponent {
+        tasks: Task[];
+        // ...
+    }
+    <!-- tab: app.component.html -->
+    <dx-gantt ... 
+        <dxo-tasks [dataSource]="tasks" [endExpr]="endDate" [startExpr]="startDate" ></dxo-tasks>
+		// ...
+    </dx-gantt>
+    <!-- tab: app.service.ts -->
+    let tasks: Task[] = [{
+        'id': 1,
+        'parentId': 0,
+        'title': 'Software Development',
+        'startDate': new Date('2019-02-21T05:00:00.000Z'),
+        'endDate': new Date('2019-07-04T12:00:00.000Z'),
+        'progress': 31
+    }, 
+    // ...   
+    ]; 
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxGantt ...
+            <DxTasks 
+                :data-source="tasksDataSource"
+                endExpr="endDate"
+                startExpr="startDate" />
+			// ...
+        </DxGantt>
+    </template>
+    <script>
+        import {
+            DxGantt,
+            DxTasks,
+			// ...
+        } from 'devextreme-vue/gantt';
+
+        import {
+            tasks,
+			// ...
+        } from './data.js';
+
+        export default {
+            components: {
+                DxGantt,
+                DxTasks,
+				// ...
+            },
+            data() {
+                return {
+                tasks,
+				// ...
+                };
+            },
+            computed: {
+                tasksDataSource() { return tasks; },
+				// ...
+            }
+        };
+    </script>
+    <!-- tab: data.js -->
+    export const tasks = [{
+        'id': 1,
+        'parentId': 0,
+        'title': 'Software Development',
+        'startDate': new Date('2019-02-21T05:00:00.000Z'),
+        'endDate': new Date('2019-07-04T12:00:00.000Z'),
+        'progress': 31
+    },
+    // ...
+    ];
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import Gantt, { Tasks, Dependencies, Resources, ResourceAssignments, Column, Editing } from 'devextreme-react/gantt';
```

Comment:
We usually show only those imports that are used in the markup. As I can see, this example uses only `Gantt` and `Tasks`.